### ISSUE TITLE
--list-software --output-format=rst fails when using python 3

### ISSUE THREAD

**Issue by migueldiascosta** [easybuilders/easybuild-framework#3112]
it fails at

https://github.com/easybuilders/easybuild-framework/blob/64f0b30d3e1b74e7bf6106e8778212c1595bed61/easybuild/tools/docs.py#L638-L645

with
```
File "/usr/lib64/python3.6/distutils/version.py", line 46, in __eq__
    c = self._cmp(other)
  File "/usr/lib64/python3.6/distutils/version.py", line 337, in _cmp
    if self.version < other.version:
TypeError: '<' not supported between instances of 'str' and 'int'
```

**Comment by migueldiascosta** [comment]
this is an old limitation with `LooseVersion` in python 3, https://bugs.python.org/issue14894, which may also affect easyblock functionality, not just documentation

In our particular case, this is currently a problem in around 20 different software packages (https://gist.github.com/migueldiascosta/d3b5620e1e1118da72d6db0e25d16bbe)

**Comment by boegel** [comment]
@migueldiascosta In #2982 we fixed this by stripping out non-digit characters, maybe that's acceptable here too?

**Comment by migueldiascosta** [comment]
9 of the cases in that list seem to be with the use of `system` as version

there are some pathological cases, e.g. with `Caffe` where 'rc3' is compared to `1.0`, and the two different versioning schemes in `OpenFOAM`

for the rest, yes, maybe stripping out the non-digit characters is an acceptable workaround

**Comment by boegel** [comment]
@migueldiascosta It took a bit of work, but a better solution (loosely based on a the proposed patch from https://bugs.python.org/issue14894) is implemented in #3146...
